### DISCUSSION TITLE
Laravel Boolean Dates

### DISCUSSION THREAD

**Discussion by sebastiaanluca** [HassanZahirnia/laravel-package-ocean#137] [Package Suggestions]
### Description

**Automatically convert Eloquent model boolean fields to dates (and back to booleans)** so you always know _when_ something was accepted or changed.

Say you've got a registration page for users where they need to accept your terms and perhaps can opt-in to certain features using checkboxes. With the new(-ish) GDPR privacy laws, you're somewhat required to not just keep track of the fact *if* they accepted those (or not), but also *when* they did.

https://github.com/sebastiaanluca/laravel-boolean-dates

### Example

User registration controller:

```php
$input = request()->input();

$user = User::create([
    'has_accepted_terms' => $input['terms'],
    'is_subscribed_to_newsletter' => $input['newsletter'],
]);
```

Anywhere else in your code:

```php
// true or false (boolean)
$user->has_accepted_terms;

// 2018-05-10 16:24:22 (Carbon instance)
$user->accepted_terms_at;
```

**Comment by HassanZahirnia** [comment]
Thanks for the suggestion 😊 I've added the package!

**Reply by sebastiaanluca** [reply]
Awesome, thanks!
